### PR TITLE
Add Celery task name and ID to log messages

### PR DESCRIPTION
**Pull request by seanh**
Add the Celery task name and ID to all log messages logged by any part
of our code or third-party code called during a Celery task.

This is so we can see which log messages come from Celery tasks in the
logs, and easily filter to all log messages from a given task.

`{task.name}[{task.id}]` (with square brackets around the ID) is used as
the format for the task name and ID in the log messages because this is
how Celery formats it when representing a task in its own log messages.
Example:
`h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268]`

`Queue` no longer needs to use Celery's `get_task_logger()` so remove
that.

Examples of how the log messages now look:

Celery logging when it receives a task. This is logged by the main
Celery process not by the task, so this log message is unchanged by this
commit:

    worker (stderr)      | [2020-10-21 14:50:45,257: INFO/MainProcess] Received task: h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268]   expires:[2020-10-21 13:51:15.251935+00:00]

The `elasticsearch` lib logging `GET`'s and `POST`'s, now with the task
name and ID:

    worker (stderr)      | [2020-10-21 14:50:45,978: INFO/ForkPoolWorker-7] h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268] GET http://localhost:9200/hypothesis/_search [status:200 request:0.029s]
    worker (stderr)      | [2020-10-21 14:50:48,959: INFO/ForkPoolWorker-7] h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268] POST http://localhost:9200/_bulk [status:200 request:0.023s]

The batch indexer logging its indexing rate, now with the task name and
ID:

    worker (stderr)      | [2020-10-21 14:50:53,705: INFO/ForkPoolWorker-7] h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268] indexed 2k annotations, rate=261/s

The job queue logging its results, now with the task name and ID:

    worker (stderr)      | [2020-10-21 14:51:14,850: INFO/ForkPoolWorker-7] h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268] {'Annotations synced because they were not in Elasticsearch': 10000}

Celery logging the task result. This unfortunately now includes the task
name and ID twice because Celery also puts them in the `%(message)s`
part of the message. I don't think there's anything to be done about
this:

    worker (stderr)      | [2020-10-21 14:51:14,888: INFO/ForkPoolWorker-7] h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268] Task h.tasks.indexer.sync_annotations[12dd9a0d-a6d3-4917-b6f9-40b47ed9f268] succeeded in 29.621645140927285s: None

### CODE REVIEW

File: h/celery.py [hunk 1]
```diff
@@ -108,3 +108,39 @@ def start(argv, bootstrap):
...
+    * Messages logged by third-party libraries that get called (directly or
+      indirectly) by @celery.task functions and that use their own loggers
+      (e.g. the elasticsearch library uses a logger called "elasticsearch")
+    """
+
+    logging.getLogger().handlers[0].setFormatter(
+        logging.Formatter(
+            "[%(asctime)s: %(levelname)s/%(processName)s] "
+            + f"{task.name}[{task_id}] "
+            + "%(message)s"
```
Comment (seanh) [inline h/celery.py:144]: This is the same as the default format (or whatever format was being used before this, anyway) but with the `f"{task.name}[{task_id}] "` inserted.

The way this works is by getting the root logger of the Celery worker process (`logging.getLogger()`) and replacing its formatter. We're assuming that the root logger has exactly one handler (`logging.getLogger().handlers[0]`), which happens to be the case.

Comment (jon-betts) [reply h/celery.py:144]: Any chance of a comment here? Or assign `root_logger = logging.getLogger().handlers[0]` or something? If we feel the need to explain this it's a good hint it's not obvious

File: h/celery.py [hunk 2]
```diff
@@ -108,3 +108,39 @@ def start(argv, bootstrap):
...
+    * Messages logged by our @celery.task functions, perhaps using Celery's
+      get_task_logger()
+
+    * Messages logged by other parts of our code that get called (directly or
+      indirectly) by @celery.task functions and that use a normal logger (e.g.
+      logging.getLogger(__name__))
+
+    * Messages logged by third-party libraries that get called (directly or
+      indirectly) by @celery.task functions and that use their own loggers
+      (e.g. the elasticsearch library uses a logger called "elasticsearch")
```
Comment (jon-betts) [inline h/celery.py:137]:
```suggestion
   * Logging by Celery itself
   * Logging in our @celery.task functions or anything they call (directly 
     or indirectly)
```

**Review by jon-betts — APPROVED** [review]
Yep looks good. Could do with a comment or something to highlight what `logging.getLogger().handlers[0]` is, also the description is a bit wordy for what it says.

File: h/celery.py (hunk 1 above)
Comment (seanh) [reply h/celery.py:148]: Changed this to:

```python
    # Replace the root logger's formatter with one that includes task.name and
    # task_id in the format. This assumes that the root logger has one handler,
    # which happens to be the case.
    root_loggers_handler = logging.getLogger().handlers[0]

    root_loggers_handler.setFormatter(
        logging.Formatter(
            "[%(asctime)s: %(levelname)s/%(processName)s] "
            + f"{task.name}[{task_id}] "
            + "%(message)s"
        )
    )
```
